polkadot/xcm/pallet-xcm/src/precompiles/IXcm.sol

```diff
@@ -1,29 +1,45 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.20;
 
-/// @title Defines all functions that can be used to interact with XCM
-/// @dev Parameters MUST use SCALE codec serialisation
+/// @dev The XCM Precompile contract address.
+address constant XCM_PRECOMPILE_ADDRESS = address(0xA0000);
+
+/// @title An interface for interacting with `pallet_xcm`
+/// @notice Provides functions for executing and sending XCM messages.
+/// Learn more about XCM: https://docs.polkadot.com/develop/interoperability/
+/// @dev All parameters must be encoded using the SCALE codec.
 interface IXcm {
-    /// Weight v2
+    /// @notice Weight v2
```

Comment:
Would be nice to link to some docs about weight here (maybe to the polkadot wiki?)